packages/transaction-controller/src/utils.ts

```diff
@@ -116,6 +145,50 @@ export function validateTxParams(
     }
   }
 }
+/**
+ * Validates the recipient address in a transaction's parameters.
+ *
+ * @param txParams - The transaction parameters object to validate.
+ * @throws Throws an error if the recipient address is invalid:
+ * - If the recipient address is an empty string ('0x') or undefined and the transaction contains data,
+ * the "to" field is removed from the transaction parameters.
+ * - If the recipient address is not a valid hexadecimal Ethereum address, an error is thrown.
+ */
+function validateRecipient(txParams: TransactionParams) {
+  if (txParams.to === '0x' || txParams.to === undefined) {
+    if (txParams.data) {
+      delete txParams.to;
+    } else {
+      throw new Error(
+        `Invalid "to" address: ${txParams.to} must be a valid string.`,
+      );
+    }
+  } else if (txParams.to !== undefined && !isValidHexAddress(txParams.to)) {
+    throw new Error(
+      `Invalid "to" address: ${txParams.to} must be a valid string.`,
+    );
+  }
+}
+/**
```

Comment:
Are these comments enforced by the linter? If not, should we remove these comments for internal functions to minimise the development effort and not state the obvious?

Reply:
It was enforced by the lint, but now that it is moved to a `validation.ts` and it's all exported it makes more sense to leave it like this even if all the functions are already concise and simple.

Reply:
I was assuming the new functions would be internal only hence me questioning the need for comments?

I think it's always nicer to minimise the exposed API where possible so there is less coupling between files and classes.